test-schema.json

```diff
@@ -27,6 +27,20 @@
                 "type": "array",
                 "items": { "$ref": "#/$defs/test" },
                 "minItems": 1
+            },
+            "specification":{
+                "description": "Location of the test case in the specification",
+                "type": "object",
+                "minProperties": 1,
+                "properties": {
+                    "jsonschema-core": { "$ref": "#/$defs/spec"},
```

Comment:
`core` and `validation` are going to be the most common values here so I think it's fine to leave off the `jsonschema`.

Also I doubt this reference (`spec`) will be useful anywhere else except within this subschema, so I'd put it here for readability personally, rather than below.